sympy/ntheory/continued_fraction.py

```diff
@@ -1,6 +1,53 @@
 from sympy.core.numbers import Integer, Rational
+from sympy.core.compatibility import as_int
+from sympy.utilities.misc import filldedent
 
 
+def continued_fraction(a):
+    """Return the continued fraction representation of a Rational or
+    quadratic irrational.
+
+    Examples
+    ========
+
+    >>> from sympy.ntheory.continued_fraction import continued_fraction
+    >>> from sympy import sqrt
+    >>> continued_fraction((1 + 2*sqrt(3))/5)
+    [0, 1, [8, 3, 34, 3]]
+
+    See Also
+    ========
+    continued_fraction_periodic, continued_fraction_reduce, continued_fraction_convergents
+    """
+    from sympy.core.singleton import S
+    from sympy.core.sympify import _sympify
+    from sympy.core.exprtools import factor_terms
+    e = _sympify(a)
+    if e.is_number:
+        n, d = e.as_numer_denom()
+        if d.is_Integer:
+            n, d = e.as_numer_denom()
+            q = None
+            if n.is_Integer:
+                c = n
+                q = 0
+            elif n.is_Add and d.is_Integer:
```

Comment:
How about the case with no rational part:
```
>>> continued_fraction(sqrt(2))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy/ntheory/continued_fraction.py", line 49, in continued_fraction
    'expecting a rational or quadratic irrational, not %s' % a)
ValueError: expecting a rational or quadratic irrational, not sqrt(2)
```

Reply:
Thanks1 It's been upgraded now and tested.